### PR TITLE
Add CRUD for card and point

### PR DESCRIPTION
**Pull request by alphashiv27**
### **PR Type**
Enhancement, Other

___

### **Description**
- Added CRUD operations for `Card` and `Point` models using Peewee ORM.
- Introduced new models `Card` and `Point` with appropriate fields and relationships.
- Added a health check endpoint `/health` to the FastAPI application.
- Configured an in-memory PostgreSQL database for the models.
- Updated project dependencies to include `peewee`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>app.py</strong><dd><code>Add health check endpoint to FastAPI application</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/app.py

<li>Added a new health check endpoint <code>/health</code>.<br> <li> Removed the previous root endpoint <code>/</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/blank-manash/feedback/pull/5/files#diff-04791d82dd15fdd480f084d7ef65a10789fa5012cb7935f76080763444d48a00">+5/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>cards_helper.py</strong><dd><code>Implement CRUD operations for Card model</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/persistence/cards_helper.py

<li>Implemented CRUD operations for <code>Card</code> model.<br> <li> Functions include create, update, read, and increase points.<br>

</details>

  </td>
  <td><a href="https://github.com/blank-manash/feedback/pull/5/files#diff-f19abe04e38994d88a20e3325753aad60e9580eaefcadb08d042caaec5944c10">+21/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>base_model.py</strong><dd><code>Define base model with Peewee ORM</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/persistence/models/base_model.py

<li>Defined a base model using Peewee ORM.<br> <li> Configured an in-memory PostgreSQL database.<br>

</details>

  </td>
  <td><a href="https://github.com/blank-manash/feedback/pull/5/files#diff-7be43cac46a858a9b72ae307d58b4c4002d306e0548ece1b9ec09bc1c452b3f7">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>card.py</strong><dd><code>Create Card model with Peewee ORM</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/persistence/models/card.py

<li>Created <code>Card</code> model with fields for ID, title, and points.<br> <li> Utilized Peewee and pgvector for field definitions.<br>

</details>

  </td>
  <td><a href="https://github.com/blank-manash/feedback/pull/5/files#diff-d522768533efcbb72926c9a744d1ca0955e83ac85401b021d58bd787869d43c5">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>point.py</strong><dd><code>Create Point model with foreign key to Card</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/persistence/models/point.py

<li>Created <code>Point</code> model with fields for ID, description, and complaints.<br> <li> Established a foreign key relationship with <code>Card</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/blank-manash/feedback/pull/5/files#diff-9425a1b9e32d00861871ef91d846a33c88e69884c72b589098b054326fc8fb99">+12/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>points_helper.py</strong><dd><code>Implement CRUD operations for Point model</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/persistence/points_helper.py

<li>Implemented CRUD operations for <code>Point</code> model.<br> <li> Functions include create, update, read, delete, and increase <br>complaints.<br>

</details>

  </td>
  <td><a href="https://github.com/blank-manash/feedback/pull/5/files#diff-23444013e9c8a1b95935d320c51b4ce7b54ebea872b82d4960d311794ee16eb2">+30/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Add Peewee ORM to project dependencies</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pyproject.toml

- Added `peewee` as a new dependency.

</details>

  </td>
  <td><a href="https://github.com/blank-manash/feedback/pull/5/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: src/persistence/models/card.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from peewee import UUIDField, TextField, BigIntegerField
+
+from src.persistence.models.base_model import BaseModel
+from pgvector.peewee import VectorField
+
+class Card(BaseModel):
+    id: UUIDField()
```
Comment (blank-manash) [inline src/persistence/models/card.py]: Can we use the autogenerated from PostGres instead?

File: src/persistence/points_helper.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    return point.save()
+
+def update_point(point_id:str,description: str,card_id:str) -> bool:
+    return (Point
+            .update(description=description,card_id=card_id)
+            .where(Point.id == point_id)
+            .execute()) == 1
+
+def increase_complaints_on_card(card_id:str) -> int:
+    return (Point
```
Comment (blank-manash) [inline src/persistence/points_helper.py:20]: Exception handling?

**Review by blank-manash — CHANGES_REQUESTED** [review]
Autogenerate ID from postgres if possible. We can create our own basemodel which has maybe all the meta fields like

```python
class BaseModel(pydantic.BaseModel):
    id, created_by, modified_by, created_at, modified_at
````

Secondly we need to implement DB exception handling and error documentation.

**Comment by alphashiv27** [comment]
> Autogenerate ID from postgres if possible. We can create our own basemodel which has maybe all the meta fields like
> 
> ```python
> class BaseModel(pydantic.BaseModel):
>     id, created_by, modified_by, created_at, modified_at
> ```
> 
> Secondly we need to implement DB exception handling and error documentation.

fuck you

File: src/persistence/points_helper.py (hunk 2 above)
Comment (alphashiv27) [reply src/persistence/points_helper.py:20]: do it yourself

File: src/persistence/models/card.py (hunk 1 above)
Comment (alphashiv27) [reply src/persistence/models/card.py]: done

**Review by blank-manash — APPROVED** [review]

**Pull request merged**